src/main/resources/avro/bdg.avdl

```diff
@@ -257,32 +257,43 @@ record AlignmentRecord {
   union { null, Contig } mateContig = null;
 }
 
+enum SequencingPlatform {
+  Illumina
+}
+
+record IlluminaInfo {
+  union { null, string } flowcellId;
+  union { null, int } lane;
+  union { null, int } tile;
+  union { null, int } surface;
+  union { null, int } xPosition;
+  union { null, int } yPosition;
+}
+
 record Sequence {
+  /**
+   Analogous to fastq
+   */
+  union { null, string } sequenceName = null;
   union { null, string } bases = null;
   union { null, string } qualities = null;
+  union { null, SequencingPlatform } platformType = null;
+  union { null, IlluminaInfo } platformInfo = null;
 }
 
 /**
-   The DNA fragment that is was targeted by the sequencer, resulting in
-   one or more reads.
+   A set of objects that should be physically analyzed/stored together.
+   Typically the result of some type of groupby operation.
```

Comment:
> I'm not entirely sure of the utility of storing a separate array<Sequence>, as you should just be able to pull the sequences from the AlignmentRecord objects. I see something like this:

I think as @ryan-williams was noting [here](
https://github.com/bigdatagenomics/adam/pull/815/files#r39452128), there's a bit of confusion around the `array<Sequence>`. To copy my comments from the other thread, I was thinking we'd null out the sequence and quality in the nested `AlignmentRecord`s and then repopulate them on-demand.